File: codeforces/training/cf810_d2_b_summer_sell_off.cpp
//
// Author: eloyhz
// Date: Sep/17/2020
//
// Codeforces (CF810-D2-B) - Summer sell-off
// https://codeforces.com/contest/810/problem/B
//

#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n, f, k, l;
    vector<tuple<int, int, int>> a;

    cin >> n >> f;
    while (n--) {
        cin >> k >> l;
        // from editorial
        a.push_back({min(k * 2, l) - min(k, l), k, l});
    }
    sort(a.begin(), a.end(), greater<tuple<int, int, int>>());
    long long ans = 0;
    int factor = 2;
    for (auto d : a)    {
        if (f > 0)  {
            f--;
        }
        else    {
            factor = 1;
        }
        ans += min(get<1>(d) * factor, get<2>(d));
    }
    cout << ans << endl;

    return 0;
}

File: codeforces/training/cf810_d2_b_summer_sell_off.py
#
# Author: eloyhz
# Date: Sep/17/2020
#
# Codeforces (CF79-D12-B) - Colorful Field
# https://codeforces.com/contest/810/problem/B
#


if __name__ == '__main__':
    n, f = [int(x) for x in input().split()]
    a = []
    for _ in range(n):
        k, l = [int(x) for x in input().split()]
        a.append([l, k])  # clients, products
    a.sort(reverse=True)
    ans = 0
    for x in a:
        if x[1] == 0:   # this day there are no products
            continue
        if f > 0:
            x[1] *= 2   # double the products the first f days
            f -= 1
        else:
            break
    for x in a:
        if x[1] == 0:
            continue
        if x[0] <= x[1]:
            ans += x[0]
        else:
            ans += x[1]
    print(ans)
